libcodechecker/analyze/log_parser.py

```diff
@@ -93,14 +93,28 @@ def parse_compile_commands_json(logfile, add_compiler_defaults=False):
     counter = 0
     for entry in data:
         sourcefile = entry['file']
+
+        if not os.path.isabs(sourcefile):
+            # Newest versions of intercept-build can create the 'file' in the
+            # JSON Compilation Database as a relative path.
+            sourcefile = os.path.join(os.path.abspath(entry['directory']),
+                                      sourcefile)
+
         lang = option_parser.get_language(sourcefile[sourcefile.rfind('.'):])
 
         if not lang:
             continue
 
         action = build_action.BuildAction(counter)
 
-        command = entry['command']
+        if 'command' in entry:
+            command = entry['command']
+        elif 'arguments' in entry:
+            # Newest versions of intercept-build create an argument vector
+            # instead of a command string.
+            command = ' '.join(entry['arguments'])
```

Comment:
The changes in this patch are specifically around creating a new architecture for the commands. There are already at least one issue about argument list escaping, #505 comes to mind. The moment we start to refurbish the `check` command, we'll look into these issues.

Right now, `ld-logger` puts custom arguments from

```bash
CodeChecker check -n run -b "g++ -DVAR='va lu e' main.cpp -o a.out"
```

to the build.json as: `\"-DVAR=va lu e\"` but these values &ndash; and I've checked it just now &ndash; do not reach the analyzer command.

In case of `intercept-build`, however, `-DVAR=va ` reaches the analyzer. I will now make sure the same escape sequence is applied to the command here, and we will discuss proper argument forwarding later.